app/validators/active_ama_validator.rb

```diff
@@ -0,0 +1,15 @@
+class ActiveAMAValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    ama = if options.key?(:comment)
```

Comment:
This logic might better be expressed as `value.respond_to?(:post) ? value.post : value` or something else along those lines, so that we don't just have flags controlling it?  At the very least I think a `for: :comment` option might be more clear than `comment: true`.  A good rule of thumb I've found is to always second-guess boolean options — would an enum be better?